### DISCUSSION TITLE
Import external model information to OpenTCS?

### DISCUSSION THREAD

**Discussion by DYD1122** [openTCS/opentcs#69] [General]
I have an idea. Can I set the starting position and ending position of the agv so that OpenTCS can read the data and move the vehicle by itself? For example, first I set up a map in OpenTCS, then I use Gurobi solver to calculate the optimal path of the vehicle and import the calculated data into OpenTCS. I want to know does this work?

**Comment by swltr** [comment]
I am not sure what data you actually want to import. You could calculate an exact description of the movement between two points (e.g. a function of a curve or a fine-grained sequence of intermediate points) and attach this information to a path (e.g. as a property). Your vehicle driver could then extract the information (e.g. from the path property) and send it to the vehicle. Alternatively, this precomputed movement description could also be stored on the vehicle side as part of a more detailed model. (After all, openTCS does not really use this information. It would only send it to the vehicle (driver).)

**Reply by DYD1122** [reply]
Thank you for your reply.I'm not very good at using English, maybe my description is not clear enough.I don't know if you've heard of the Gurobi solver. It's a tool that can be optimized.My goal is to test the speed of scheduling AGVs with different algorithms.If I manually set the start position, end position and path of the vehicle in OpenTCS. That there will be a bit of trouble.So I want to use Gurobi solver to first calculate various information (such as vehicle location and path) and import it into OpenTCS.Let the vehicle automatically read the data and run. I don't know if OpenTCS can do that.

**Reply by swltr** [reply]
The details of what you are trying to do are still unclear to me. What would the output of your solver be? Would it be a route from point A to point Z, possibly passing multiple points in your plant model? Or would it be a description of how exactly the vehicle moves/steers between two neighboring points?

Maybe an example would help here.
